docs/spec_draft.md

```diff
@@ -1450,6 +1451,88 @@ More formally, `result[i0, ..., iR-1] = operand[j0, ..., jR-1]` where
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.sort
+
+### Semantics
+
+Sorts a variadic number of tensors in `inputs` together, according to a custom 
+`comparator`, along the given `dimension` and produces a variadic number of
+tensors as `results`. If `is_stable` is true, then the sorting is stable, that
+is, relative order of elements considered to be equal by the comparator is
+preserved.
+
+More formally, for all `0 <= id < jd < di`, either 
+`compare_i_j = compare_j_i = false` or `compare_i_j = true`, where:
+  1. `compare_i_j` $=$ `comparator(inputs[0][i], inputs[0][j], inputs[1][i], inputs[1][j], ...)`.
+  1. For all indices `i = [i0, ..., iR-1]` and `j = [j0, ..., jR-1]`.
+  1. Where `i` $=$ `j` everywhere except for the `d`th dimension.
+  1. Where `d` $=$ `dimension >= 0 ? dimension : rank(inputs[0]) + dimension`.
+
+### Inputs
+
+| Name         | Type                                              |
+|--------------|---------------------------------------------------|
+| `inputs`     | variadic number of tensors of any supported types |
+| `dimension`  | `si64`                                            |
+| `is_stable`  | `i1`                                              |
+| `comparator` | `function`                                        |
+
+### Results
+
+| Name      | Type                                              |
+|-----------|---------------------------------------------------|
+| `results` | variadic number of tensors of any supported types |
+
+### Constraints
+
+  * (C1) `inputs` have alteast 1 tensor.
+  * (C2) `results` have same size as `inputs`.
+  * (C3) All tensors in `inputs` and `results` have the same shape.
+  * (C4) `-R` $\le$ `dimension` $\lt$ `R`, where `R` is rank of `inputs[0]`.
+  * (C5) `comparator` has type `(E1, E1, ..., EN-1, EN-1) -> i1`, where `Ei` is
+         element type of `inputs[i]`.
+
+### Examples
+
+```mlir
+// Sort along dimension 0
+
+// %input0 = [[1,2,3],[3,2,1]]
+// %input1 = [[3,2,1],[1,2,3]]
+%result0, %result1 = "stablehlo.sort"(%input0, %input1) ({
+  ^bb0(%arg0: tensor<i32>, %arg1: tensor<i32>, %arg2: tensor<i32>, %arg3: tensor<i32>):
+    %predicate = "stablehlo.compare"(%arg0, %arg1) {
+      comparison_direction = #stablehlo<comparison_direction GT>
+      } : (tensor<i32>, tensor<i32>) -> tensor<i1>
+    "stablehlo.return"(%predicate) : (tensor<i1>) -> ()
+}) {
+  dimension = 0 : i64,
+  is_stable = true
+  } : (tensor<2x3xi32>, tensor<2x3xi32>) -> (tensor<2x3xi32>, tensor<2x3xi32>)
```

Comment:
This line needs to be deindented to match the style of other ops where attributes are split across multiple lines. (Here and below).